src/pages/thought_modal/thought_modal.ts

```diff
@@ -1,28 +1,50 @@
 import { Component } from '@angular/core';
-import { IonicPage, ViewController } from 'ionic-angular';
+import { IonicPage, ViewController, ToastController } from 'ionic-angular';
+import { ThoughtsProvider } from '../../providers/thoughts/thoughts';
 
-/**
- * Generated class for the ThoughtPage page.
- *
- * See https://ionicframework.com/docs/components/#navigation for more info on
- * Ionic pages and navigation.
- */
 
 @IonicPage()
 @Component({
   selector: 'page-thought_modal',
   templateUrl: 'thought_modal.html',
 })
+
 export class ThoughtModalPage {
+  thought = {}
+
+  constructor(private view: ViewController,
+              public thoughtsProvider: ThoughtsProvider,
+              private toastCtrl: ToastController) {
+  }
+
+  presentToast(msg) {
+    let toast = this.toastCtrl.create({
+      message: msg,
+      duration: 2500,
+      position: 'top'
+    });
 
-  constructor(private view: ViewController) {
+    toast.present();
   }
 
   closeModal() {
     this.view.dismiss();
   }
 
+  thoughtCreate(flash_msg) {
+    if (flash_msg == "Thought was successfully created.") {
+      this.closeModal();
+    }
+    this.presentToast(flash_msg)
+  }
+
   ionViewDidLoad() {
   }
 
+  createThought() {
+    this.thoughtsProvider.saveThought(this.thought)
+    .subscribe(data => this.thoughtCreate("Thought was successfully created."),
+              data => this.thoughtCreate(data.json().error[0]))
```

Comment:
~~Why are we calling `this.thoughtCreate` again here? ~~

~~This is supposed to be the `onError` where you handle any errors that occur while executing the request. I would expect to see something along the lines of~~

```javascript
.subscribe(
  data => doSomethingOnSuccess(),
  error => doSomethingWithError()
)
```

Perhaps find a better name for `thoughtCreate()` -- I only realized half way through my comment that you actually use that to display various messages to the user

Reply:
something like this is probably better

```javascript
.subscribe(
  data => {
    this.presentToast("Thought was successfully created.");
    this.closeModal();
  },
  error => this.presentToast(getErrorMessageFrom(error))
)
```

IMO You don't really need that `thoughtCreate()` method

Reply:
Fixed : )